stage #2.5/module #4 REST Basics - Spring Web/#1.REST Basics/content.md

```diff
@@ -0,0 +1,479 @@
+# REST Basics - Spring Web
+
+## Basics of RESTful APIs
+_**REST is an API architecture style.**_ It stands for Representational State Transfer (REST). 
+RREST is an architectural style that defines a set of constraints to be used for creating web services. 
+REST API is a way of accessing web services in a simple and flexible way without having any processing.
+REST technology is generally preferred to the more robust Simple Object Access Protocol (SOAP) technology because REST uses less bandwidth, 
+simple and flexible making it more suitable for internet usage. It’s used to fetch or give some information from a web service. 
+All communication done via REST API uses only HTTP request.
+
+## Richardson REST API Maturity Model
+The Richardson Maturity Model grades RESTful API by their maturity. The REST Maturity Model is a way to grade your API according to the constraints of REST.
+It describes four different levels of REST (starting from 0 to 3 level). A REST API that supports hypermedia controls is classified as Level 3 in this maturity model.
+- **Level 0 - a single URI and use a single HTTP method (typically POST):** <br> These services have a single URI and use a single HTTP method (typically POST). These are the most primitive way of building SOA applications with a single POST method and using XML to communicate between services.Level zero of maturity does not make use of any of URI, HTTP Methods, and HATEOAS capabilities.
```

Comment:
'These are the most primitive way...' should be changed to 'This is the most primitive way...'
'...to communicate between services.Level zero of...' - missing space before 'Level'